app/src/main/java/org/oppia/android/app/onboardingv2/CreateProfileFragmentPresenter.kt

```
@@ -0,0 +1,122 @@
+  private val fragment: Fragment,
+  private val activity: AppCompatActivity,
+  private val createProfileViewModel: CreateProfileViewModel
+) {
+  private lateinit var binding: CreateProfileFragmentBinding
+  private lateinit var uploadImageView: ImageView
+  private var selectedImage: Uri? = null
+
+  /** Initialize layout bindings. */
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View {
+    binding = CreateProfileFragmentBinding.inflate(
+      inflater,
+      container,
+      /* attachToRoot= */ false
+    )
+    binding.let {
+      it.lifecycleOwner = fragment
+      it.viewModel = createProfileViewModel
+    }
+
+    uploadImageView = binding.createProfileUserImageView
+    Glide.with(activity)
+      .load(R.drawable.ic_default_avatar)
+      .listener(object : RequestListener<Drawable> {
+        override fun onLoadFailed(
+          e: GlideException?,
+          model: Any?,
+          target: Target<Drawable>?,
+          isFirstResource: Boolean
+        ): Boolean {
+          return false
+        }
+
+        override fun onResourceReady(
+          resource: Drawable?,
+          model: Any?,
+          target: Target<Drawable>?,
+          dataSource: DataSource?,
+          isFirstResource: Boolean
+        ): Boolean {
+          uploadImageView.setColorFilter(
+            ResourcesCompat.getColor(
+              activity.resources,
+              R.color.component_color_avatar_background_25_color,
+              null
+            ),
+            PorterDuff.Mode.DST_OVER
+          )
+          return false
+        }
+      })
+      .into(uploadImageView)
+
+    binding.onboardingNavigationContinue.setOnClickListener {
+      val nickname = binding.createProfileNicknameEdittext.text.toString().trim()
+
+      if (nickname.isNotBlank()) {
+        createProfileViewModel.hasError.set(false)
+      } else {
+        createProfileViewModel.hasError.set(true)
+      }
```

It would be nice to hide the error message when there is change in text in the edit text, instead of hiding it when the user clicks on the `Continue` button again.

Since the error message says,
> Click in the box above to type your nickname.

Alternatively we could just hide the error message when the user clicks on the edit text to type the nickname (i.e., the edit text regains focus).

---

Added a text watcher to listen for text change.

---

Please add tests to verify this behavior.

---

Added `testFragment_onTextChanged_afterError_hidesErrorMessage`